app/screens/edit_profile/edit_profile.tsx

```diff
@@ -0,0 +1,481 @@
+        }, componentId);
+
+        return () => {
+            unsubscribe.remove();
+        };
+    }, [userInfo]);
+
+    useEffect(() => {
+        const backHandler = BackHandler.addEventListener('hardwareBackPress', close);
+        return () => {
+            backHandler.remove();
+        };
+    }, []);
+
+    useEffect(() => {
+        if (!isTablet) {
+            setButtons(componentId, {
+                rightButtons: [rightButton!],
+                leftButtons: [leftButton!],
+            });
+        }
+    }, []);
+
+    const close = useCallback(() => {
+        if (isModal) {
+            dismissModal({componentId});
+        } else if (isTablet) {
+            DeviceEventEmitter.emit(Events.ACCOUNT_SELECT_TABLET_VIEW, '');
+        } else {
+            popTopScreen(componentId);
+        }
+
+        return true;
+    }, []);
+
+    const enableSaveButton = useCallback((value: boolean) => {
+        if (!isTablet) {
+            const buttons = {
+                rightButtons: [{
+                    ...rightButton!,
+                    enabled: value,
+                }],
+            };
+            setButtons(componentId, buttons);
+        }
+        setCanSave(value);
+    }, [componentId, rightButton]);
+
+    const submitUser = useCallback(preventDoubleTap(async () => {
+        enableSaveButton(false);
+        setError(undefined);
+        setUpdating(true);
+        try {
+            const partialUser: Partial<UserProfile> = {
+                email: userInfo.email,
+                first_name: userInfo.firstName,
+                last_name: userInfo.lastName,
+                nickname: userInfo.nickname,
+                position: userInfo.position,
+                username: userInfo.username,
+            };
+
+            const {error: reqError} = await updateMe(serverUrl, partialUser);
+
+            if (reqError) {
+                resetScreen(reqError as Error);
+                return;
+            }
+
+            close();
+        } catch (e) {
+            resetScreen(e as Error);
+        }
+    }), [userInfo]);
+
+    const resetScreen = useCallback((resetError: Error) => {
+        setError(resetError?.message);
+        Keyboard.dismiss();
+        setUpdating(false);
+        enableSaveButton(true);
+        scrollViewRef.current?.scrollToPosition(0, 0, true);
+    }, [enableSaveButton]);
+
+    const updateField = useCallback((fieldKey: string, name: string) => {
+        const update = {...userInfo};
+        update[fieldKey] = name;
+        setUserInfo(update);
+
+        // @ts-expect-error access object property by string key
+        const currentValue = currentUser[fieldKey];
+        const didChange = currentValue !== name;
+        enableSaveButton(didChange);
+    }, [userInfo, currentUser]);
+
+    const includesSsoService = (sso: string) => ['gitlab', 'google', 'office365'].includes(sso);
+    const isSAMLOrLDAP = (protocol: string) => ['ldap', 'saml'].includes(protocol);
```

Comment:
Nit: These could be defined outside the component. On this particular case, the only advantage of doing it is my personal feeling that it makes things clearer, since it shows that this does not depend on anything (on any prop, or any state...) and makes it easier to avoid variable clashes and mistakes.

Reply:
Refactored.